plugins/endpoints/src/main/java/org/pytorch/serve/plugins/endpoint/Token.java

```diff
@@ -0,0 +1,188 @@
+
+@Endpoint(
+        urlPattern = "token",
+        endpointType = EndpointTypes.MANAGEMENT,
+        description = "Token authentication endpoint")
+public class Token extends ModelServerEndpoint {
+    private static String apiKey;
+    private static String managementKey;
+    private static String inferenceKey;
+    private static Instant managementExpirationTime;
+    private static Instant inferenceExpirationTime;
+    private static Integer timeToExpiration = 30;
+    private SecureRandom secureRandom = new SecureRandom();
+    private Base64.Encoder baseEncoder = Base64.getUrlEncoder();
+
+    @Override
+    public void doGet(Request req, Response rsp, Context ctx) throws IOException {
+        String queryResponse = parseQuery(req);
+        String test = "";
+        if ("management".equals(queryResponse)) {
+            generateKeyFile(1);
+        } else if ("inference".equals(queryResponse)) {
+            generateKeyFile(2);
+        } else {
+            test = "{\n\t\"Error\": " + queryResponse + "\n}\n";
+        }
+        rsp.getOutputStream().write(test.getBytes(StandardCharsets.UTF_8));
+    }
+
+    // parses query and either returns "management"/"inference" or a wrong type error
+    public String parseQuery(Request req) {
+        QueryStringDecoder decoder = new QueryStringDecoder(req.getRequestURI());
+        Map<String, List<String>> parameters = decoder.parameters();
+        List<String> values = parameters.get("type");
+        if (values != null && !values.isEmpty()) {
+            if ("management".equals(values.get(0)) || "inference".equals(values.get(0))) {
+                return values.get(0);
+            } else {
+                return "WRONG TYPE";
+            }
+        }
+        return "NO TYPE PROVIDED";
+    }
+
+    public String generateKey() {
+        byte[] randomBytes = new byte[6];
+        secureRandom.nextBytes(randomBytes);
+        return baseEncoder.encodeToString(randomBytes);
+    }
+
+    public Instant generateTokenExpiration(Integer time) {
+        return Instant.now().plusSeconds(TimeUnit.MINUTES.toSeconds(time));
+    }
+
+    // generates a key file with new keys depending on the parameter provided
+    // 0: generates all 3 keys
+    // 1: generates management key and keeps other 2 the same
+    // 2: generates inference key and keeps other 2 the same
+    public boolean generateKeyFile(Integer keyCase) throws IOException {
+        String fileData = " ";
+        String userDirectory = System.getProperty("user.dir") + "/key_file.txt";
+        File file = new File(userDirectory);
+        if (!file.createNewFile() && !file.exists()) {
+            return false;
+        }
+        if (apiKey == null) {
+            apiKey = generateKey();
+        }
+        switch (keyCase) {
+            case 1:
+                managementKey = generateKey();
+                managementExpirationTime = generateTokenExpiration(timeToExpiration);
```

Comment:
timeToExpiration is class variable, why give it as a parameter?

Reply:
Fixed